### PR TITLE
fix(travis): Make code pass lint and improve the travis configuration.

### PR DESCRIPTION
**Pull request by martinstreicher**
@colinrymer @Teapane This is ready for a review. This should get us to the point where a semantic release is sent to npm. It does not yet deploy to S3.

Some questions:

The latest release was 1.2.2. Can we make the first release to npm 1.3.0 since the current code has a new feature in it. How do we set the baseline version number for semantic-release to use?

Once a release to npm occurs, I can likely find the version number of that release and use those numbers to perform the release to S3. Previously, the version number came from package.json as it was a literal string. With semantic-release, there is a placeholder and the ultimate version number is left in npm. I'll take those numbers and a modified deploy.js script and push to S3. OK?

### CODE REVIEW

**Comment by martinstreicher** [comment]
@colinrymer @Teapane All this was running fine on Travis until I split the PR. :-(

**Comment by martinstreicher** [comment]
A `git diff` against travis_configuration, the branch associated with the closed PR, shows only code changes for the feature. :-1:

**Comment by colinrymer** [comment]
Unless this needs to be merged today, let's hold off on this until tomorrow when I'm back in front of a computer normally (currently traveling and dealing with :shit: internet). Once I'm back to a normal internet connection, I'll work on debugging why Travis is failing, etc. @martinstreicherif you want to schedule a block of time tomorrow (Friday, 2015/03/18) to pair on this, I'm more than happy to do that (if possible, including @Teapane would be great).

**Comment by martinstreicher** [comment]
It can wait. Not sure why everything was humming today and then it broke. Makes no sense.

**Comment by Teapane** [comment]
Boom :boom:

**Comment by Teapane** [comment]
Regarding your comment on the latest release. I *think* since this is the first time it is published to NPM, it will release it as `v1.0.0` and semvar on from there.

**Comment by martinstreicher** [comment]
I am OK with the v1.0.0 release. We'll just reset some of the stuff in S3 and the documentation to suit.

**Comment by martinstreicher** [comment]
So Travis is back to normal with no intervention. A review can proceed. I rebased against master since @mattp88 pulled in a feature PR from another branch.

**Comment by Teapane** [comment]
Travis was using an outdated version of `npm`, FYI.

**Comment by martinstreicher** [comment]
How did it get fixed? Was it a configuration in our account?

**Comment by Teapane** [comment]
I pushed a commit. `npm` was installing `v2.3.0`, but then we overrode it with a global install of `v2.0.0`, which broke some libs. I just removed that install, and default to the newest version.

**Comment by martinstreicher** [comment]
Right. Thanks. That code for npm 2.0.0 was added by the generated code you committed a few steps back. No worries. Just glad its resolved.

File: .travis.yml [hunk 1]
```diff
@@ -23,12 +23,9 @@ notifications:
 
 sudo: false
 
-
-#
 # the process
 
 before_install:
-  - npm i -g npm@^2.0.0
```
Comment (martinstreicher) [inline .travis.yml]: Thanks!

**Comment by colinrymer** [comment]
:+1: LGTM

**Comment by Teapane** [comment]
LGTM :+1:

**Comment by martinstreicher** [comment]
Thanks.

**Pull request merged**